sdk-java/src/main/java/com/spotify/confidence/FlagResolverClientImpl.java

```diff
@@ -24,8 +36,20 @@ public CompletableFuture<ResolveFlagsResponse> resolveFlags(
           openFeatureEvaluationContext.getStructValue().getFieldsMap());
       evaluationContextBuilder.removeFields(OPEN_FEATURE_RESOLVE_CONTEXT_KEY);
     }
+    if (telemetry != null) {
+      telemetry.setIsProvider(isProvider);
+    }
```

Comment:
I am a bit concerned about this: flipping a variable on each resolveFlags call sounds tricky to get right. I am also thinking that this variable refers to the current resolve call, but the latency data in Monitoring is about a set of resolve calls that could have been spawned from provider or non-provider

Reply:
Yeah, I swapped the way of doing this but this requires the end users to build the Confidence instance with `buildForProvider()` :/

Reply:
I believe now we have both a `isProvider` argument in resolve and a `buildForProvider` builder method. The first is used for our legacy Telemetry, the latter for the new one. I think we should strive for at least having one solution for both. But I am not blocking the PR with this, given the "best effort" approach to the new Telemetry work

Reply:
I'll follow this up in a separate PR.